File: Jin/week4/day18/튜플.java
package Jin.week4.day18;

import java.util.*;

public class 튜플 {
    public int[] solution(String s) {
        // 입력값을 String[] 로 관리
        s = s.substring(2, s.length()- 2).replace("},{", "-");
        String[] sArr = s.split("-");

        // length 기준으로 정렬
        Arrays.sort(sArr, Comparator.comparingInt(String::length));

        int[] answer = new int[sArr.length];
        int index = 0;
        for (String str : sArr) {
            String[] check = str.split(",");

            for (int i = 0; i < check.length; i++) {
                int num = Integer.parseInt(check[i]);

                // 배열 내에 값이 존재하지 않는다면 배열에 추가
                boolean exists = Arrays.stream(answer).anyMatch(n -> n == num);
                if (!exists) {
                    answer[index++] = num;
                }
            }
        }

        return answer;
    }
}

File: Jiho/week4/day18/튜플.js
function solution(s) {
  const answer = [];
  const tuples = [];

  for (const tuple of s.split("},")) {
    tuples.push(
      tuple
        .replaceAll("{", "")
        .replaceAll("}", "")
        .split(",")
        .map((v) => Number(v))
    );
  }

  tuples.sort((a, b) => a.length - b.length);

  for (const tuple of tuples) {
    answer.push(...tuple.filter((v) => !answer.includes(v)));
  }

  return answer;
}
